File: scripts/generate_index.py
#!/usr/bin/env python3
"""
Generate Report Index

data/reports/ ディレクトリ内のJSONファイルをスキャンし、
data/index.json を生成する。
"""

import json
import os
import glob
from datetime import datetime

def generate_index():
    base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
    reports_dir = os.path.join(base_dir, "data", "reports")
    index_file = os.path.join(base_dir, "data", "index.json")
    
    # JSONファイルを取得
    report_files = glob.glob(os.path.join(reports_dir, "*.json"))
    
    # 日付リストを作成（ファイル名から抽出）
    dates = []
    for filepath in report_files:
        filename = os.path.basename(filepath)
        date_str = os.path.splitext(filename)[0]
        try:
            # 日付形式のチェック（YYYY-MM-DD）
            datetime.strptime(date_str, "%Y-%m-%d")
            dates.append(date_str)
        except ValueError:
            print(f"Skipping invalid filename: {filename}")
            continue
            
    # 新しい順にソート
    dates.sort(reverse=True)
    
    # index.json を書き出し
    with open(index_file, "w", encoding="utf-8") as f:
        json.dump({"dates": dates, "updatedAt": datetime.now().isoformat()}, f, ensure_ascii=False, indent=2)
        
    print(f"Generated index.json with {len(dates)} reports.")
    print(f"Path: {index_file}")

if __name__ == "__main__":
    generate_index()

File: scripts/generate_index.js
#!/usr/bin/env node

/**
 * Generate Report Index
 *
 * data/reports/ ディレクトリ内のJSONファイルをスキャンし、
 * data/index.json を生成する。
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function generateIndex() {
  const baseDir = path.join(__dirname, '..');
  const reportsDir = path.join(baseDir, 'data', 'reports');
  const indexFile = path.join(baseDir, 'data', 'index.json');

  // JSONファイルを取得
  const files = fs.readdirSync(reportsDir).filter(f => f.endsWith('.json'));

  // 日付リストを作成（ファイル名から抽出、YYYY-MM-DD形式のみ）
  const dates = [];
  for (const file of files) {
    const dateStr = path.basename(file, '.json');
    if (/^\d{4}-\d{2}-\d{2}$/.test(dateStr)) {
      dates.push(dateStr);
    } else {
      console.log(`Skipping invalid filename: ${file}`);
    }
  }

  // 新しい順にソート
  dates.sort((a, b) => b.localeCompare(a));

  // index.json を書き出し
  const index = { dates, updatedAt: new Date().toISOString() };
  fs.writeFileSync(indexFile, JSON.stringify(index, null, 2), 'utf8');

  console.log(`Generated index.json with ${dates.length} reports.`);
  console.log(`Path: ${indexFile}`);
}

generateIndex();
